src/QirRuntime/test/unittests/QirRuntimeTests.cpp

```diff
@@ -14,34 +14,33 @@
 #include "qsharp__core__qis.hpp"
 #include "QirRuntime.hpp"
 
-#include "BitStates.hpp"
 #include "QirContext.hpp"
 #include "SimulatorStub.hpp"
 
 using namespace Microsoft::Quantum;
 
 struct ResultsReferenceCountingTestQAPI : public SimulatorStub
 {
-    int lastId = 1;
+    int lastId = -1;
     const int maxResults;
-    BitStates allocated;
+    std::vector<bool> allocated;
 
     static int GetResultId(Result r)
     {
         return static_cast<int>(reinterpret_cast<int64_t>(r));
     }
 
     ResultsReferenceCountingTestQAPI(int maxResults)
-        : maxResults(maxResults + 2)
+        : maxResults(maxResults)
     {
-        allocated.ExtendToInclude(maxResults);
+        allocated = std::vector<bool>(maxResults);
     }

```

Comment:
I expect this fragment to default-construct the values in the vector. 
The type `bool` is a built-in type. For the built-in types the default-construction is implemented by means of "doing nothing", i.e. leaving the items *un*initialized. 
I expect the values of the vector to stay uninitialized after this constructor. Consider changing to something like this
```c++
    ResultsReferenceCountingTestQAPI(int maxResults)
        : maxResults(maxResults)
        , allocated(maxResults, false)    // Init all elems to false.
    {
    }
```

Reply:
I could've sworn that I saw somewhere that the default constructor for `bool` always produced a value of `false` but now I can't find it, and the few explanations I can find say that the details are platform/implementation dependent. You are right, it would be much better to be explicit and use the syntax you suggest than leave it up to the individual behavior of the underlying language implementation, especially if that can vary.